R/cube-dims.R

```diff
@@ -144,29 +144,84 @@ setMethod("[", "CubeDims", function (x, i, ...) {
     return(CubeDims(x@.Data[i], names=x@names[i]))
 })
 
-is.selectedDimension <- function (dims) {
-    is.it <- function (x, dim, MRaliases) {
-        maybe <- x$alias %in% MRaliases &&
-            x$type == "categorical" &&
-            length(dim$name) == 3
-        if (maybe) {
-            cats <- Categories(data=x$categories)
-            ## Unlike the strict is.3vl, this doesn't compare cat names because
-            ## they've already been munged to TRUE/FALSE
-            maybe <- setequal(ids(cats), c(-1, 0, 1)) &&
-                sum(is.selected(cats)) == 1 &&
-                sum(is.na(cats)) == 1
+is.selectedDimension <- function (dims) getDimTypes(dims) == "mr_selections"
+
+#' Get dimension type
+#'
+#' This function returns the specific type of each cube dimension. This is useful
+#' when cubes contain categorical array or multiple response variables because it
+#' identifies the dimensions of the cube which refer to the different parts of
+#' array variable:
+#' - `ca_items`: Categorical array items
+#' - `ca_categories`: The categories of the categorical array
+#' - `mr_items`: Multiple response options or items
+#' - `mr_selections`: The selection status for a multiple response variable
+#'
+#' @return A character vector. This is identical to `types()` except that
+#' the array variable types are more specific.
+#' @param x a CrunchCube or CubeDims object
+#'
+#' @return a character vector of dimension types
+#' @export
+#' @keywords internal
+getDimTypes <-  function (x) {
```

Comment:
Why does this function need to be exported?

Reply:
It doesn't really. There's a similar version being used in crplyr, that we anticipated removing in deference to this one. But we could always use `crunch:::getDimTypes` there since we control both packages.

Reply:
We can't `:::` if we want it on CRAN, so I guess that means we need to export.